backend/routers/battle.js

```diff
@@ -38,6 +39,119 @@ async function getUsersOnPlanet(req, res, next) {
   }
 }
 
+function getCasualtyPercentage(troop, targetTroop, correction = 1) {
+  return (targetTroop.attack / (targetTroop.attack + troop.attack))
+    * ((troop.defence + 100) / (2 * troop.defence + 100))
+    * (0.6 + 0.8 * Math.random())
```

Comment:
what are these magic numbers and what are they doing exactly?

Reply:
The function with 100 calculates the reduction in loss caused by the defence. The loss will decrease with the defence increasing but never reaches zero percent. The 0.6 and 0.8 means 1+_0.4, which add randomness to the battle.

Reply:
please add this as a comment, because this explanation noone will know what are these numbers and why are they there